src/pybind/mgr/dashboard/rbd_ls.py

```diff
@@ -2,7 +2,47 @@
 import rados
 import rbd
 from remote_view_cache import RemoteViewCache
+from types import VersionedSyncObject
 
+class RbdPoolLs(RemoteViewCache):
+    def __init__(self, module_inst, osd_pools):
+        super(RbdPoolLs, self).__init__(module_inst)
+        self.osd_pools = osd_pools
+        self.rbd_pools = []
+
+    def _get(self):
+        from mgr_module import ceph_state
+        ctx_capsule = ceph_state.get_context()
+
+        self.log.info("Constructing Rados")
+        cluster = rados.Rados(context=ctx_capsule)
+        cluster.connect()
+
+        for pool in self.osd_pools:
+            self.log.info("Constructing IOCtx " + pool)
+            try:
+                ioctx = cluster.open_ioctx(pool)
+                ioctx.stat("rbd_directory")
+                self.rbd_pools.append(pool)
+            except (rados.PermissionError, rados.ObjectNotFound):
+                self.log.info("No RBD directory in " + pool)
+            except:
+                self.log.error("Failed to open pool " + pool)
+        return self.rbd_pools
+
+class RbdPools(VersionedSyncObject):
```

Comment:
Yeah -- I couldn't for the life of me figure out how it was being being used. I just figured it was a "TODO" so if I could just add code to cache RBD pools per epoch, it's one less thing to change if and when it does cache.